app/interactors/book_suggestions_by_genre.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+class BookSuggestionsByGenre
+  include Interactor
+
+  def call
+    context.fail!(message: 'Book no has genre') unless book.genre
+    context.books = Book.where(genre: book.genre).where.not(id: book_ids)
+  end
+
+  private
+
+  def book_ids
+    user.rents.pluck(:book_id).uniq
+  end
+
+  def user
+    @user ||= context.user
+  end
+
+  def book
+    context.fail!(message: 'Book id is null') unless context.book_id
```

Comment:
Another way of handling this sort of things is letting it fail. `Book#find` will raise an exception (`ActiveRecord::RecordNotFound`) if no Book could be found. A very common practice is to rescue from that error in APIController and return a 404 response instead of checking the existence/validity of each param.
(What you did here is fine, you don't have to change nothing!)